browser/lib/util/http.ts

```diff
@@ -0,0 +1,176 @@
+import * as Utils from '../../../common/lib/util/utils';
+import Defaults from '../../../common/lib/util/defaults';
+import ErrorInfo from '../../../common/lib/types/errorinfo';
+import { IHttp, RequestCallback, RequestParams } from '../../../common/types/http';
+import HttpMethods from '../../../common/constants/HttpMethods';
+
+type Rest = any;
+type Realtime = any;
+
+const noop = function() {};
+
+function shouldFallback(err: ErrorInfo) {
+	const statusCode = err.statusCode as number;
+	/* 400 + no code = a generic xhr onerror. Browser doesn't give us enough
+		* detail to know whether it's fallback-fixable, but it may be (eg if a
+		* network issue), so try just in case */
+	return (statusCode === 408 && !err.code) ||
```

Comment:
If HTTP status codes aren't available from one of our dependencies then perhaps we could pull them out to our own module somewhere?